### PR TITLE
[MAD][DATA1018][lab-intro-to-ml] Miriam Montaño

### PR DESCRIPTION
**Pull request by miriammg**
Hola Belén, 
Práctica muy interesante, esta muy bien guiada y explicada, me parece buena idea reforzar la limpieza de datasets con nulos, cambiar tipos, fechas y demás. En el bonus 2 he aplicado las funciones básicas según el lab. Mañana, en un hueco, podemos ver confusion_matrix y precision_score. No me ha quedado muy claro el tema de kernel lineal vs. radial. 

Un saludo,

### CODE REVIEW

File: module-3/lab-intro-to-ml/your-code/main.ipynb [hunk 1]
```diff
@@ -12,7 +12,7 @@
   },
```
Comment (Gobinde43) [inline module-3/lab-intro-to-ml/your-code/main.ipynb]: - "Let's examine the DewPointAvgF variable by using the unique() function to list all unique values in this dataframe."
- "Pandas lo toma como un objeto supongo que porque solo existen 66 valores únicos y son muy repetidos. Lo toma como una variable categórica." En realidad es porque hay un "-" en los datos y no puede identificar el tipo de objeto adecuadamente. 
- The following using apply can be something like the
```Python
for e in wrong_type_columns: 
    austin[e] = pd.to_numeric(austin[e], errors='coerce')

#using apply 
austin[wrong_type_columns] = austin[wrong_type_columns].apply(lambda col: pd.to_numeric(col, errors='coerce'))
```
-" Next, count the number of rows of austin and missing_values." Try the following and check the differences between len and size methods in a dataframe. 
```Python
print(austin.size)
print(missing_values.size)
```
- "As you can see from the output, the majority of missing data is in one column called PrecipitationSumInches. What's the number of missing values in this column in ratio to its total number of rows?" You have done the following: 
```Python
b = ((austin['PrecipitationSumInches'].isna().sum())/(len(austin)))*100
#when you should have done something similar to the following
missing_values['PrecipitationSumInches'].size / austin['PrecipitationSumInches'].size
```
- "What is the largest number of events happened in a single day? Enter your answer in the next cell." Applying a head() to austin_fixed would give you the largest number. 
```Python
austin_fixed.head()
```
- Challenge 4: Other way of processing the date column
```Python
from datetime import datetime as dt
austin_fixed['Date'] = austin_fixed['Date'].apply(lambda date_col: dt.strptime(date_col, '%Y-%m-%d').toordinal())
```
- "Your y should be a subset of austin_fixed containing one column TempAvgF." Try the following instead of writing down each element of X vector and re-train the model. 
```Python
response_col_names = list(austin_fixed.columns.values)
response_col_names.remove('TempAvgF')
X = austin_fixed[response_col_names]
y = austin_fixed['TempAvgF']
```
- If you want to play with synthetic data for classification problems, scikit-learn has a function to generate synthetic datasets. https://scikit-learn.org/stable/modules/generated/sklearn.datasets.make_classification.html.
It's very useful to play with different type of classifiers (two label)
```Python
from sklearn.datasets import make_classification
````
- Try to train your SVM both linear and rbf, check the differences in the performance metric. 
- The readme says at the end of the lab, "You now see the model fit score of your test data set. If it's extremely low, it means your selected model is not a good fit and you should try other models." Your performance metric is not so low, try to think about what's going on.
